books/RayTracingInOneWeekend.html

```diff
@@ -1446,12 +1457,195 @@
     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ C++
             return 0.5 * (rec.normal + color(1,1,1));
         }
+
+        vec3 unit_direction = unit_vector(r.direction());
+        auto a = 0.5*(unit_direction.y() + 1.0);
+        return (1.0-a)*color(1.0, 1.0, 1.0) + a*color(0.5, 0.7, 1.0);
+    }
     ...
     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     [Listing [main-with-interval]: <kbd>[main.cc]</kbd> The new main using interval]
 
 
 
+A New Camera Class
+====================================================================================================
+Before continuing, now is a good time to consolidate our camera and scene-render code into a single
+  new class: `camera`.
+The camera class will be responsible for constructing and dispatching rays into the world, and using
+  the results to construct the rendered image.
+
+In this refactoring, we'll pull in the `ray_color()` function along with the image, camera and
+  render sections of our main program.
+The new camera class will contain two public methods: `initialize()` and `render()`, plus two
+  private helper methods `get_ray()` and `ray_color()`.
+
+Ultimately, the camera will use a bunch of public variables as parameters instead of using a
+  constructor with a long sequence of arguments, or using a bunch of setter methods.
+It's crude, but simple.
+Given this, the usage pattern will be such that the main code will create a default camera,
+  then set different parameters (perhaps at multiple places in the code).
+When everything's been set, it will call the `render()` method.
+The `render()` method will set up the camera by calling the `initialize()` function, and then
+  running the render loop.
+
+Here's the skeleton of the new `camera` class:
+
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ C++
+    #ifndef CAMERA_H
+    #define CAMERA_H
+
+    #include "rtweekend.h"
+
+    #include "color.h"
+    #include "hittable.h"
+
+    class camera {
+      public:
+        /* Public Camera Parameters Here */
+
+        void render(const hittable& world) {
+        }
+
+      private:
+        /* Private Camera Variables Here */
+
+        void initialize() {
+        }
+
+        color ray_color(const ray& r, const hittable& world) const {
+            return color();
```

Comment:
replace with `...` or a comment or emptyness

I'd rather that the skeleton doesn't compile correctly.

Reply:
Done. Used ellipses.